### ISSUE TITLE
Python API gdal.Translate generated incorrect data while splitting a geotiff file

### ISSUE THREAD

**Issue by ruanqizhen** [OSGeo/gdal#7160]
## Expected behavior and actual behavior.
[updated]

Tested with source file https://prd-tnm.s3.amazonaws.com/StagedProducts/Elevation/1m/Projects/NY_CMPG_2013/TIFF/USGS_one_meter_x58y451_NY_CMPG_2013.tif

I created the out.tif file with command gdal_translate -projwin 585000 4509999 585099 4509900 source.tif out.tif
and created the out1.tif with python method: gdal.Translate(out_file, ds, options="-b 1 -projwin 585000 4509999 585099 4509900")

Then I tried to read the data back with Python:

dataset = gdal.Open(source_file, gdal.GA_ReadOnly)
band = dataset.GetRasterBand(1)
data = band.ReadAsArray()
For out.tif, I got array([[11.339995 , 11.389996 , 11.370011 , ... which are expected;
For out1.tif, I got array([[-3.4028235e+38, -3.4028235e+38, -3.4028235e+38, ..., all elements are equal to nodata_value.

## Steps to reproduce the problem.
See above

## Operating system
Ubuntu 16.04 64 bit

## GDAL version and provenance

3.0.12

**Comment by rouault** [comment]
Please provide the source dataset

**Comment by ruanqizhen** [comment]
The source file I tested was: https://prd-tnm.s3.amazonaws.com/StagedProducts/Elevation/1m/Projects/MA_NE_CMGP_Sandy_Z19_A2_2015/TIFF/USGS_one_meter_x31y468_MA_NE_CMGP_Sandy_Z19_A2_2015.tif

**Comment by rouault** [comment]
This file has no explicit mask, but just a nodata value, so it is expected you won't get an explicit mask in the output.
You could ask for materializing an explicit mask with
```
gdal.Translate(out_file, ds, options = "-b 1 -mask mask -projwin 585000 4509999 585099 4509900")
```

**Comment by ruanqizhen** [comment]
I did some further evaluation on this problem, it was not a problem of mask, problem was that the data in the generated file was in correct. 
I'm still tested with file  https://prd-tnm.s3.amazonaws.com/StagedProducts/Elevation/1m/Projects/NY_CMPG_2013/TIFF/USGS_one_meter_x58y451_NY_CMPG_2013.tif

I created the out.tif file with command `gdal_translate -projwin 585000 4509999 585099 4509900 a3.tif out.tif`
and created the out1.tif with python method: `gdal.Translate(out_file, ds, options="-b 1 -projwin 585000 4509999 585099 4509900")`

Then I tried to read the data back with Python:
```
dataset = gdal.Open(source_file, gdal.GA_ReadOnly)
band = dataset.GetRasterBand(1)
data = band.ReadAsArray()
```

For out.tif, I got `array([[11.339995 , 11.389996 , 11.370011 , ...` which are expected;
for for out1.tif, I got `array([[-3.4028235e+38, -3.4028235e+38, -3.4028235e+38, ...`, all elements are equal to nodata_value.

**Comment by jratike80** [comment]
This works for me. GDAL 3.7.0dev-0febc574fa, released 2023/01/22 from OSGeo4W and Python 3.9.5

```
from osgeo import gdal
input=gdal.Open("USGS_one_meter_x58y451_NY_CMPG_2013.tif")
output = gdal.Translate("out.tif", input, options="-b 1 -projWin 585000 4509999 585099 4509900")
test=gdal.Open("out.tif")
band=test.GetRasterBand(1)
data=band.ReadAsArray()
print(data)
[[11.339995  11.389996  11.370011  ... 11.780003  11.690005  11.6300125]
 [11.310004  11.399985  11.439992  ... 11.720004  11.650013  11.670003 ]
 [11.359999  11.370005  11.419988  ... 11.670013  11.690002  11.709997 ]
 ...
 [11.089978  11.119982  11.159983  ... 11.739997  11.739999  11.739999 ]
 [10.969981  10.999983  11.059981  ... 11.729997  11.729999  11.729999 ]
 [10.849982  10.889983  10.959979  ... 11.719997  11.72      11.720003 ]]
```
